source/Armament.cpp

```diff
@@ -233,7 +233,8 @@ void Armament::Aim(const FireCommand &command)
 
 // Fire the given weapon, if it is ready. If it did not fire because it is
 // not ready, return false.
-void Armament::Fire(int index, Ship &ship, vector<Projectile> &projectiles, vector<Visual> &visuals, bool jammed)
+void Armament::Fire(int index, Ship &ship, vector<Projectile> &projectiles,
```

Comment:
Is there any particular reason you did it here instead of doing is as part of the projectile?

Reply:
because I do not want to spawn a projectile, i want a ship

Reply:
Yeah but why cant a projectile spawn a ship thats my question

Reply:
I do think that spawning ships should be similar to spawning submunitions- on the death of the projectile. This makes it more versatile. If you want the current behavior, give the projectile a lifetime of 0.

Reply:
Hmmm, I get that, there are some barriers in doing that, but I guess I can do it with some work

Reply:
done